### PR TITLE
protocol: api to register schemes that can handle service worker

### PR DESCRIPTION
**Pull request by deepak1556**
https://github.com/atom/electron/issues/2831 is now completely addressed. Have enabled service worker for file scheme by default. For privileged custom schemes users can use the api.

### CODE REVIEW

File: atom/renderer/atom_renderer_client.cc [hunk 1]
```diff
@@ -129,6 +130,9 @@ void AtomRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
   new PepperHelper(render_frame);
   new AtomRenderFrameObserver(render_frame, this);
+
+  // Allow file scheme to handle service worker by default.
+  blink::WebSecurityPolicy::registerURLSchemeAsAllowingServiceWorkers("file");
```
Comment (zcbenz) [inline atom/renderer/atom_renderer_client.cc]: This seems redundant since we are already adding `file` protocol in `AddServiceWorkerSchemes`?

Comment (deepak1556) [inline atom/renderer/atom_renderer_client.cc]: Its required to bypass the security restrictions of blink layer, `AddServiceWorkerSchemes` is only used on the browser side by `ServiceWorker` request handler.

Comment (zcbenz) [inline atom/renderer/atom_renderer_client.cc]: Then shouldn't other schemed also be added here?

Comment (deepak1556) [inline atom/renderer/atom_renderer_client.cc]: we have `webframe::registerURLSchemeAsPrivileged` to do that. So users who need custom schemes to handle service workers have to use both these api to allow on both render and browser side.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**